store/store_action.go

```diff
@@ -556,6 +567,18 @@ func (s *Store) snapAction(ctx context.Context, currentSnaps []*CurrentSnap, act
 	}
 
 	if resp.StatusCode != 200 {
+		// some fields might not be supported on proxies with old versions.
+		// we should retry with the snap store version known as we can now
+		// get it from the response header.
+		if resp.StatusCode == 400 && storeVer <= 0 {
```

Comment:
what about the error we were getting even with 54? was it a 400 or a 500? do we need to fallback to turn 54 into 53 perhaps if we reach here with that error combination? the question I suppose is whether there is a stable proxy version shipped with the problem or not yet?

Reply:
a different approach would to ask the store to bump the version (to 55) once all the issues are solved and use that here instead

Reply:
The solution was intended to be generic so other features might use in the future for different api versions, I can't think of a way that can be used to determine which API version to fallback to in case of failure with api known.
I like the deterministic nature of the version bump approach.